MapboxMobileEvents/MMEEventsManager.m

```diff
@@ -70,22 +76,28 @@ - (void)dealloc {
 
 - (void)initializeWithAccessToken:(NSString *)accessToken userAgentBase:(NSString *)userAgentBase hostSDKVersion:(NSString *)hostSDKVersion {
     self.apiClient = [[MMEAPIClient alloc] initWithAccessToken:accessToken userAgentBase:userAgentBase hostSDKVersion:hostSDKVersion];
+    self.configurationUpdater.delegate = self;
     
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(pauseOrResumeMetricsCollectionIfRequired) name:UIApplicationDidEnterBackgroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(pauseOrResumeMetricsCollectionIfRequired) name:UIApplicationDidBecomeActiveNotification object:nil];
+    void(^initialization)(void) = ^{
```

Comment:
I think we should do the `weakself`/`strongself` dance here, and then return early if `strongself` is `nil`.  My concern is the timer manager at the bottom of the block and any potential retain cycles.